fsspec/compression.py

```diff
@@ -139,7 +139,7 @@ def _fetch_range(self, start, end):
 try:
     import snappy
 
-    snappy.compress
+    snappy.compress(b"")
```

Comment:
Wait, this one isn't quite right, it's an attribute look up that could be an assert statement I suppose

Reply:
Indeed, I guess the idea was that the `compress(b"")` call cannot fail; any error has to be an attribute look up. I can revert to the pure attribute look up with `# noqa`, it had the merit to be more explicit.

Reply:
An assert in regular code is not recommended, since they can be turned off for the whole interpreter session. I think the function call is fine, or if we wanted a different function call to appease linting, we could do a `getattr`.

Background: `snappy-python` is not the only installable to provide a package called `snappy` (not my fault!), so this is a way to differentiate. Actually, snappy-compressing whole files (as opposed to buffers within files) is super-rare, so we could perhaps even remove this, rather than increase import time by checking this. We should probably not be importing anything at import time!

```
In [1]: import fsspec
In [2]: import sys
In [3]: "snappy" in sys.modules
Out[3]: True
```

I intend to merge this PR as is; @DimitriPapadopoulos or @Skylion007 , you are most welcome to make a PR to defer imports in the fsspec.compression module.